### ISSUE TITLE
RUN-315: Syncing LDAP User Information Is Not Working

### ISSUE THREAD

**Issue by paulcalabro** [rundeck/rundeck#7175]
_Note: Bug Reports not following this template may be closed without feedback._

**Getting Help**

**Describe the bug**

I am enabling the LDAP sync feature for a Docker container via these environment variables:

- RUNDECK_JAAS_LDAP_USERFIRSTNAMEATTRIBUTE: givenName
- RUNDECK_JAAS_LDAP_USERIDATTRIBUTE: sAMAccountName
- RUNDECK_JAAS_LDAP_USERLASTNAMEATTRIBUTE: sn

These appear to be inserted into the correct configuration files as expected, however, the syncing of LDAP information is not occurring:

```
sh-4.4$ grep sync /home/rundeck/server/config/rundeck-config.properties
...
# user sync
rundeck.security.syncLdapUser=true
...
```

```
sh-4.4$ grep -E Attribute  /home/rundeck/server/config/jaas-loginmodule.conf
...
        userLastNameAttribute="sn"
        userFirstNameAttribute="givenName"
        userEmailAttribute="mail"
...
```

**My Rundeck detail**

* Rundeck version: 3.4.1-20210715 
* install type: docker
* OS Name/version: Ubuntu 18.04.5 LTS
* DB Type/version: mariadb

**To Reproduce**
1. Perform an LDAP search and confirm the attributes are correct.
2. Set the appropriate environment variables.
3. Verify settings appear in files generated from templates.
4. Log in to Rundeck.
5. View user profile information and observe they are blank.

**Expected behavior**
I expected to see my first name, last name, and company email address.

**Screenshots**
![Screen Shot 2021-07-20 at 9 13 08 PM](https://user-images.githubusercontent.com/585059/126429771-c83fa1e9-e2cd-4dff-80ce-ec7d96d19a96.png)

**Desktop (please complete the following information):**
 - OS: macOS 11.4
 - Browser Firefox 89

**Additional context**
N/A

**Comment by mlachal** [comment]
Hi, I just observed the same issue on our 3.4.1 RPM platform. The syncLdapUser functionnality works fine on 3.3.11 and it looks like the issue is there : https://github.com/rundeck/rundeck/commit/c228eb199c92dd178c8b2908d30f299e956f7404#diff-f2e0d41b63fe1c660b65a51f9d507689ac2c1465ae0431ea1ce64b9ff29e5c00R41

**Comment by paulcalabro** [comment]
@mlachal Thanks for the info. I also observed the same behavior on 3.4.0 as well.

**Comment by ChBerrich** [comment]
Hi, me too, same issue since 3.4.0 migration, works fine on 3.3.13

**Comment by MarcusCaepio** [comment]
Hi, same for Active Directory and 3.4.1

**Comment by FedeHAL** [comment]
Hi, same error with 3.4.2-20210803 rpm version

**Comment by MegaDrive68k** [comment]
Hi folks,

Could you try using the `RUNDECK_SECURITY_SYNCLDAPUSER` parameter?

I tested using the following docker-compose file (from [this](https://github.com/rundeck/docr-zoo/tree/master/ldap) example) and it works if you add the email property on each user on the [LDIF](https://github.com/rundeck/docker-zoo/blob/master/ldap/ldif/50-bootstrap.ldif) file.

```yaml
version: '3'

services:
    rundeck:
        image: rundeck/rundeck:3.4.2
        environment:
            RUNDECK_JAAS_MODULES_0: JettyCombinedLdapLoginModule
            RUNDECK_JAAS_LDAP_PROVIDERURL: ldap://ldap:389
            RUNDECK_JAAS_LDAP_BINDDN: cn=admin,dc=rdtest,dc=com
            RUNDECK_JAAS_LDAP_BINDPASSWORD: AdminPass123
            RUNDECK_JAAS_LDAP_USERBASEDN: ou=users,dc=rdtest,dc=com
            RUNDECK_JAAS_LDAP_ROLEBASEDN: ou=roles,dc=rdtest,dc=com
            RUNDECK_SECURITY_SYNCLDAPUSER: "true"
            RUNDECK_JAAS_LDAP_USEREMAILATTRIBUTE: mail
        ports:
          - 4440:4440
    ldap:
        image: osixia/openldap:1.5.0
        environment:
          - LDAP_ORGANISATION=RD Test
          - LDAP_DOMAIN=rdtest.com
          - LDAP_ADMIN_PASSWORD=AdminPass123
        volumes:
          - ./ldif:/container/service/slapd/assets/config/bootstrap/ldif/custom:rw
        ports:
          - "389:389"
        command: --copy-service
```

The LDIF file modified:

```ldif
# Define top-level entry:
#dn: {{ LDAP_BASE_DN }}
#objectClass: dcObject
#objectClass: organization
#o: Example, Inc.
#dc: example

# Define an entry to contain users:
dn: ou=users,{{ LDAP_BASE_DN }}
objectClass: organizationalUnit
ou: users

# Define some users:
dn: cn=admin, ou=users,{{ LDAP_BASE_DN }}
uid: admin
userPassword: admin
objectClass: person
objectClass: top
objectClass: inetOrgPerson
sn: The admin account for the Example client to use
cn: admin
mail: admin@rdtest.com
givenName: admin first name

dn: cn=user, ou=users,{{ LDAP_BASE_DN }}
uid: user
userPassword: user
objectClass: person
objectClass: top
objectClass: inetOrgPerson
sn: Added user account
cn: user

dn: cn=build, ou=users,{{ LDAP_BASE_DN }}
uid: build
userPassword: {MD5}sNonVSCRjiPdYV4qdHUo8Q==
#####
# MD5 creds, Base64 encoded
#userPassword: build
objectClass: person
objectClass: top
objectClass: inetOrgPerson
sn: The account to use to demonstrate managing builds only
cn: build

dn: cn=deploy, ou=users,{{ LDAP_BASE_DN }}
uid: deploy
userPassword: {CRYPT}de01JmlU8XXTQ
#####
# CRYPT creds
#userPassword: deploy
objectClass: person
objectClass: top
objectClass: inetOrgPerson
sn: The account to use to demonstrate managing deployment only
cn: deploy

dn: cn=test, ou=users,{{ LDAP_BASE_DN }}
uid: test
userPassword: test
objectClass: person
objectClass: top
objectClass: inetOrgPerson
sn: Has no role access
cn: test

# Define an entry to contain roles:
dn: ou=roles, {{ LDAP_BASE_DN }}
objectClass: organizationalUnit
ou: roles

# Define some roles and their membership:
dn: cn=architect, ou=roles,{{ LDAP_BASE_DN }}
objectClass: groupOfUniqueNames
uniqueMember: cn=admin,ou=users,{{ LDAP_BASE_DN }}
cn: architect

dn: cn=admin, ou=roles,{{ LDAP_BASE_DN }}
objectClass: groupOfUniqueNames
uniqueMember: cn=admin,ou=users,{{ LDAP_BASE_DN }}
cn: admin

dn: cn=user, ou=roles,{{ LDAP_BASE_DN }}
objectClass: groupOfUniqueNames
uniqueMember: cn=admin,ou=users,{{ LDAP_BASE_DN }}
uniqueMember: cn=deploy,ou=users,{{ LDAP_BASE_DN }}
uniqueMember: cn=build,ou=users,{{ LDAP_BASE_DN }}
cn: user

dn: cn=build, ou=roles,{{ LDAP_BASE_DN }}
objectClass: groupOfUniqueNames
uniqueMember: cn=admin,ou=users,{{ LDAP_BASE_DN }}
uniqueMember: cn=build,ou=users,{{ LDAP_BASE_DN }}
cn: build

dn: cn=deploy, ou=roles,{{ LDAP_BASE_DN }}
objectClass: groupOfUniqueNames
uniqueMember: cn=admin,ou=users,{{ LDAP_BASE_DN }}
uniqueMember: cn=deploy,ou=users,{{ LDAP_BASE_DN }}
cn: deploy
```

On 3.4.2:

![Screenshot_9](https://user-images.githubusercontent.com/51376003/129278146-f9f2ab98-616d-4bcd-a274-1790d2c86738.png)

On 3.3.13:

![Screenshot_8](https://user-images.githubusercontent.com/51376003/129278149-717a621d-b0de-4bf5-a4b6-7568136d01d6.png)

Hope it helps!

**Comment by davama** [comment]
Same here.
using `rundeck-3.4.3.20210823-1.noarch`  on rocky linux 8

```
grep -i ldap /etc/rundeck/rundeck-config.properties 
# ldap
rundeck.security.syncLdapUser=true
```
 and ldap config
 ```
grep user /etc/rundeck/jaas-ldap.conf 
      bindDn="cn=manager,dc=bla,dc=bla"
      bindPassword="bla"
      userBaseDn="ou=People,dc=bla,dc=bla"
      userRdnAttribute="uid"
      userIdAttribute="uid"
      userPasswordAttribute="userPassword"
      userObjectClass="posixAccount"
      userLastNameAttribute="sn"
      userFirstNameAttribute="givenName"
      userEmailAttribute="mail"
```

no issue authenticating, just sync does not happen...

Thanks

**Comment by davama** [comment]
Just to update here:
this is still not working on `rundeck-3.4.4.20210920-1.noarch`

Thanks

**Comment by matthias-colt** [comment]
Not fixed in 3.4.6

**Comment by mathiasaerts** [comment]
Could it be this simple? https://github.com/mathiasaerts/rundeck/commit/859a7ee1534949e665f19b3169a9b37c11c3d963

**Comment by davama** [comment]
@mathiasaerts didnt try your patch but i thought it was fixed with 3.4.8 with patch https://github.com/rundeck/rundeck/commit/d8b9495b9c17874fd8c3c3e93fcf4a37deb8e465

But i guess i misunderstood that patch.

either way, this issue still present on 3.4.8

**Comment by mathiasaerts** [comment]
I was actually unable to run this myself as well. I tried to set up a test environment and run the gradle build but unfortunately I cannot get it to work and I don't have more time I can spend on this.

When searching through the code, it seems that the `rundeck.` prefix is never included in any `configurationService.get*` function call, so it seems misplaced here as a leftover from the refactor from `grailsApplication.config.*`, where @mlachal referred to here: https://github.com/rundeck/rundeck/issues/7175#issuecomment-884160843

**Comment by hdep** [comment]
Hi,

I have the same issue with 3.4.9.20211221-1

**Comment by davama** [comment]
For those who have the infra, we decided to use our http auth server instead: https://docs.rundeck.com/docs/administration/security/authentication.html#preauthenticated-mode-using-headers

We use authelia as our authoritative app: https://github.com/authelia/authelia

name,email,groups all synced fine with http headers.

**Comment by matthias-colt** [comment]
The new release [3.4.10](https://github.com/rundeck/rundeck/releases/tag/v3.4.10) seems to fix this issue (just tested on my sever).
The related PR #7434 only mentions the Rundeck enterprise (pro) release.

**Comment by rklomp** [comment]
I can confirm this is now working on community edition 3.4.10

**Comment by allenjeong** [comment]
I confirmed this is fixed on 3.4.10, but it's back on v4.0.0.

**Comment by matthias-colt** [comment]
Hi @allenjeong, I suggest to open a new ticket for v4.0.0 and reference this one.
I haven't tested 4.0.0 yet since I faced some general issue with LDAP login.
